### PR TITLE
[Loader 2 of 2] Add loader component

### PR DESCRIPTION
**Pull request by mlunoe**
---
⚠️ Followup PR to this: https://github.com/dcos/dcos-ui/pull/1007

---

This PR introduces a Loader component to be used around the application, centralizing code in one place. It is also exposing the component as a module to be used by plugins!

### CODE REVIEW

File: src/js/components/Loader.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  cubeTransition: {className: 'cube-transition', divCount: 2},
+  lineScale: {className: 'line-scale', divCount: 5},
+  lineScaleParty: {className: 'line-scale-party', divCount: 4},
+  lineScalePulseOut: {className: 'line-scale-pulse-out', divCount: 5},
+  lineScalePulseOutRapid:
+    {className: 'line-scale-pulse-out-rapid', divCount: 5},
+  lineSpinFadeLoader: {className: 'line-spin-fade-loader', divCount: 8},
+  pacman: {className: 'pacman', divCount: 5},
+  semiCircleSpin: {className: 'semi-circle-spin', divCount: 1},
+  triangleSkewSpin: {className: 'triangle-skew-spin', divCount: 1}
```
Comment (jfurrow) [inline src/js/components/Loader.js]: Do we need all of these? It would be nice to reduce these to just the few that we actually use.

Comment (mlunoe) [inline src/js/components/Loader.js]: I just thought it would be nice to have the possibilities as a reference here. What bothers you about it?

Comment (jfurrow) [inline src/js/components/Loader.js]: It seems superfluous because we will likely never use most of these; it's extra code with no benefit.

Comment (mlunoe) [inline src/js/components/Loader.js]: Right, however a good reference. We could link to the repository instead? Or comment out the ones we don't use?
It was just annoying to me to have to look this up and understand it again. That will not be the case next time we want to add a loader.

Comment (jfurrow) [inline src/js/components/Loader.js]: That's understandable. I guess I'd prefer to remove them and link to the repository of available options, but I don't care that much — if you'd prefer to comment them out, let's do that!

Comment (mlunoe) [inline src/js/components/Loader.js]: Done!

File: src/js/pages/jobs/JobDetailPage.js [hunk 2]
```diff
@@ -260,13 +261,8 @@ class JobDetailPage extends mixin(StoreMixin, TabsMixin) {
 
   getLoadingScreen() {
     return (
-      <div className="container container-fluid container-pod text-align-center
```
Comment (wavesoft) [inline src/js/pages/jobs/JobDetailPage.js]: #nit but perhaps use the ```{`string`}``` approach since it spans on multiple lines?

**Comment by wavesoft** [comment]
Health is never populated on the nodes grid. It always stays on loading state:

image

**Comment by mlunoe** [comment]
Rebased, and [this](https://github.com/dcos/dcos-ui/pull/1016#issuecomment-243709207) problem is fixed.

File: src/js/pages/jobs/JobDetailPage.js (hunk 2 above)
Comment (mlunoe) [inline src/js/pages/jobs/JobDetailPage.js]: You are commenting on lines I removed

**Comment by wavesoft** [comment]
LGTM

**Comment by mlunoe** [comment]
ping @daiyiwhale

File: src/js/components/Loader.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+class Loader extends React.Component {
+  getDivs(length) {
+    return Array.from({length}).map(function (_, index) {
+      return <div key={index} />;
+    });
+  }
+
+  render() {
+    let {className, innerClassName, type} = this.props;
```
Comment (rcorral) [inline src/js/components/Loader.js]: You don't define `type` in the proptypes.

**Pull request merged**